packages/vm/src/static_analysis.rs

```diff
@@ -0,0 +1,204 @@
+use parity_wasm::elements::{deserialize_buffer, Internal, Module};
+use std::collections::HashSet;
+
+use crate::errors::{VmError, VmResult};
+
+pub const REQUIRED_IBC_EXPORTS: &[&str] = &[
+    "ibc_channel_open",
+    "ibc_channel_connect",
+    "ibc_channel_close",
+    "ibc_packet_receive",
+    "ibc_packet_ack",
+    "ibc_packet_timeout",
+];
+
+pub fn deserialize_wasm(wasm_code: &[u8]) -> VmResult<Module> {
+    deserialize_buffer(&wasm_code).map_err(|err| {
+        VmError::static_validation_err(format!(
+            "Wasm bytecode could not be deserialized. Deserialization error: \"{}\"",
+            err
+        ))
+    })
+}
+
+pub fn exported_functions(module: &Module) -> HashSet<String> {
+    module
+        .export_section()
+        .map_or(HashSet::default(), |export_section| {
+            export_section
+                .entries()
+                .iter()
+                .filter_map(|entry| match entry.internal() {
+                    Internal::Function(_) => Some(entry.field().to_string()),
+                    _ => None,
+                })
+                .collect()
+        })
+}
+
+/// Returns true if and only if all IBC entry points ([`REQUIRED_IBC_EXPORTS`])
+/// exist as exported functions. This does not guarantee the entry points
+/// are functional and for simplicity does not even check their signatures.
```

Comment:
fair enough. if they use the `#[entry_point]` directive, the compiler asserts the function signature. If someone uploads garbage, it will bind a port and fail on usage. Same as if they implement the proper signature and `panic` there. This is not a security hole and better to just use a good approximation that works for all "proper" contract developers